solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/LTRScoringAlgorithm.java

```diff
@@ -169,5 +198,29 @@ public abstract Explanation explain(LeafReaderContext context, int doc,
   public String toString() {
     return  getClass().getSimpleName() + "(name="+getName()+")";
   }
+  
+  /**
+   * Goes through all the stored feature values, and calculates the normalized
+   * values for all the features that will be used for scoring.
+   */
+  public float[] normalizeFeatures(float[] modelFeatureValues) {
+    float[] modelFeatureValuesNormalized = modelFeatureValues.clone();
```

Comment:
Just noticed that this clone will generate garbage, and it is called per doc.  Can we get away without cloning?  We can update the comments saying that it does in place updates instead